fpdf/fpdf.py

```diff
@@ -2135,12 +2136,27 @@ def _perform_page_break(self):
         if ws > 0:
             self.ws = 0
             self._out("0 Tw")
-        self.add_page(same=True)
+        self._break_or_add_page()
         self.x = x  # restore x but not y after drawing header
         if ws > 0:
             self.ws = ws
             self._out(f"{ws * self.k:.3f} Tw")
 
+    def _break_or_add_page(self):
+        """
+        This method is useful when it is necessary to go to the next page
+        instead of creating a new one
+        """
+        if self._has_next_page():
+            self.page += 1
+        else:
+            self.add_page(same=True)
+        self.total_pages = max(self.page, self.total_pages)
+        self.set_y(self.t_margin)
```

Comment:
This could cause backward incompatible behaviours.
I'd prefer to avoid setting `.y` at this point in `FPDF` code,
unless you have a good justification

Reply:
The justification is: in case it has next page and needs to break it, the `self.y` attribute does not updates itself, and the result is this file:
[break_page_without_set_y.pdf](https://github.com/PyFPDF/fpdf2/files/7468504/break_page_without_set_y.pdf)

I will change the code, then it will only update the `self.y` attribute when it increases the page number, and not when it creates a new one.